apiserver/common/addresses.go

```diff
@@ -46,6 +47,19 @@ func (api *APIAddresser) APIHostPorts() (params.APIHostPortsResult, error) {
 	}, nil
 }
 
+// APIHostPortsForAgents returns the API server addresses that are suitable for
+// agent use based on the configured management space.
+// +// If no such space is configured, this will return the same as APIHostPorts().
+func (api *APIAddresser) APIHostPortsForAgents() (params.APIHostPortsResult, error) {
```

Comment:
Arguably if we want to add a method to the Client facade, we should be updating the version number. However, as mentioned before, we don't really need to add this method to Client.

As for the other places that are calling APIHostPorts...
I wonder if we want to add a method, or if we just change the existing method to know that it is on a facade that is used by an agent, and not a Client, and thus return the right value.
In one sense, I like adding the method, because it makes the State => apiserver => client => worker chain more obvious, as they all end up using the same named thing.
On the other hand, having to update all the worker code to use a new method, and never use the old method makes me think we really just want a single method.
Thoughts?